File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int ans = 0, sz = bank.size(), sz2 = bank[0].size();
        int prev = 0, curr = 0;
        for(int i=0; i<sz; i++){
            if(curr != 0) prev = curr;
            curr = 0;
            for(int j=0; j<sz2; j++){
                if(bank[i][j] == '1'){
                    ans += prev;
                    curr++;
                }
            }
        }
        return ans;
    }
};

File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int m = bank.length, n = bank[0].length();
        int cnt = 0, prev = 0, ans = 0;

        for(int i=0; i<m; i++){
            cnt = 0;
            for(int j=0; j<n; j++){
                if(bank[i].charAt(j) == '1') cnt++;
            }
            // System.out.println(cnt + " " + prev);
            if(cnt > 0){
                ans += cnt * prev;
                prev = cnt;
            }
        }

        return ans;
    }
}